### PR TITLE
native: fix warning errors freebsd

### PR DESCRIPTION
**Pull request by thomaseichinger**
Fixes warnings on FreeBSD which became errors in #3319.

### CODE REVIEW

File: cpu/native/native_cpu.c [hunk 1]
```diff
@@ -119,7 +119,7 @@ char *thread_stack_init(thread_task_func_t task_func, void *arg, void *stack_sta
         err(EXIT_FAILURE, "thread_stack_init: getcontext");
     }
 
-    p->uc_stack.ss_sp = stk;
+    p->uc_stack.ss_sp = (char*) stk;
```
Comment (gebart) [inline cpu/native/native_cpu.c]: This is defined as `void *ss_sp;` on Linux, this cast may introduce additional warnings about alignment. Any ideas?
Do we need an ifdef?

Comment (thomaseichinger) [inline cpu/native/native_cpu.c]: Hm, would guess `#ifdef` it is then.

Comment (authmillenon) [inline cpu/native/native_cpu.c]: normally you can cast `void *` pointers from and to anything... That's the point of `void *` pointers: being generic.

Comment (authmillenon) [inline cpu/native/native_cpu.c]: Also keep to the coding conventions: `(char *)stk;`

Comment (gebart) [inline cpu/native/native_cpu.c]: After looking through the code some more I think that stk should be changed into a `void *`

Comment (thomaseichinger) [inline cpu/native/native_cpu.c]: @authmillenon The problem is, on FreeBSD it is not a `void *` but `char *` pointer.

**Comment by thomaseichinger** [comment]
updated addressing comments.

**Comment by gebart** [comment]
@thomaseichinger how about changing the stk variable into a `void *` or `char *`?
The whole pointer math part around `#ifdef NATIVESPONTOP` seem to assume `stk` is a `void **`

**Comment by thomaseichinger** [comment]
@authmillenon could you check these commits tackle the warnings we talked about?

**Comment by authmillenon** [comment]
Yes, they are.

**Comment by thomaseichinger** [comment]
@gebart I adapted `stk` and remove `NATIVESPONTOP` as it seems it is never used. @LudwigOrtmann  can you verify this.

**Comment by gebart** [comment]
I'm assuming that the nativespontop is something that have been an experiment with upward growing stack.

**Comment by gebart** [comment]
I have some ideas for a refactoring with creating stack arrays as unsigned long instead in order to get rid of some Wextra warnings,but that is outside of the scope of this PR.

**Comment by gebart** [comment]
ACK, go after Travis is fine and @LudwigOrtmann has had the opportunity to respond to the question by @thomaseichinger

**Comment by thomaseichinger** [comment]
Got the ok of @LudwigOrtmann on a side channel.

File: cpu/native/native_cpu.c [hunk 2]
```diff
@@ -96,24 +96,18 @@ void thread_print_stack(void)
 
 char *thread_stack_init(thread_task_func_t task_func, void *arg, void *stack_start, int stacksize)
 {
-    unsigned int *stk;
+    char *stk;
```
Comment (LudwigKnuepfer) [inline cpu/native/native_cpu.c]: This broke native running on arm.

**Pull request merged**